Source/WebKit/Shared/Cocoa/CoreIPCNSValue.serialization.in

```diff
@@ -0,0 +1,37 @@
+# Copyright (C) 2023 Apple Inc. All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+# 1.  Redistributions of source code must retain the above copyright
+#     notice, this list of conditions and the following disclaimer.
+# 2.  Redistributions in binary form must reproduce the above copyright
+#     notice, this list of conditions and the following disclaimer in the
+#     documentation and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS'' AND
+# ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS BE LIABLE FOR
+# ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+#if PLATFORM(COCOA)
+
+webkit_platform_headers: "CoreIPCNSValue.h" <Foundation/NSRange.h>
+
+[WebKitPlatform] struct _NSRange {
+    NSUInteger location;
+    NSUInteger length;
+}
+
+[WebKitPlatform] class WebKit::CoreIPCNSValue {
+    std::optional<WebKit::CoreIPCNSValue::WrappedValue> m_wrappedValue;
```

Comment:
Shouldn't this be a std::variant<WebKit::CoreIPCNSValue::WrappedValue, WebKit::CoreIPCSecureCoding> since you will always only have one or the other?

Reply:
s/Shouldn't/Couldn't/

But yes, agreed, it can be!

Reply:
If they were both null/nullopt then an assertion would fail and we would dereference null.  Using std::variant would eliminate this possibility.

Reply:
Yah, I made that change (while still tracking down TV/Watch build failures)